KOGITO-5893: Data-index API connection with runtimes tasks: add comme…

nmirasch opened the pull request:
…nts and attachments

jira: https://issues.redhat.com/browse/KOGITO-5893

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

nmirasch commented:
Jenkins retest this

cristianonicolai commented on data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java:
```diff
@@ -209,27 +231,36 @@ protected CompletableFuture sendDeleteClientRequest(WebClient webClient, String
+        HttpRequest<Buffer> request = webClient.post(requestURI)
+                .putHeader("Authorization", getAuthHeader())
+                .putHeader("Content-Type", contentType);
+        if ("application/json".equals(contentType)) {
```
there must be a constant for this :)

nmirasch replied:
yep! I found that :)

cristianonicolai commented on data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:
```diff
@@ -123,8 +123,10 @@ private GraphQLSchema createSchema() {
                     builder.dataFetcher("JobReschedule", this::rescheduleJob);
-                    builder.dataFetcher("TaskUpdate", this::updateUserTask);
-                    builder.dataFetcher("TaskPartialUpdate", this::partialUpdateUserTask);
+                    builder.dataFetcher("TaskInstanceUpdate", this::updateUserTask);
```
```suggestion
                    builder.dataFetcher("UserTaskInstanceUpdate", this::updateUserTask);
```

nmirasch replied:
done

nmirasch commented:
@cristianonicolai renamed mutations and removed hard update. Replaced taskInfo parameter by the enumeration of available fields to update. Can you take a look when you have the chance? Thanks!

radtriste commented on data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/api/KogitoRuntimeClientImpl.java:
```diff
@@ -176,24 +180,31 @@ protected WebClientOptions getWebClientToURLOptions(String targetHttpURL) {
-        return sendPutClientRequest(getWebClient(serviceURL),
-                requestURI,
-                "Update UserTask: " + userTaskInstance.getName() + " with id: " + userTaskInstance.getId(),
+        return sendPatchClientRequest(getWebClient(serviceURL), requestURI,
```
changing from put to patch ?

nmirasch replied:
@radtriste not really. I've removed the old updateTask that uses the 'put' and only keep the 'partial update' that uses the patch. The difference is the put operation replaces all the fields related to the task and if they don't come in the variables the will be set to empty.. With this approach, only the parameters that came will be updated, it allows more control over what is affected by the update operation.

radtriste commented on data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:
```diff
@@ -123,8 +123,9 @@ private GraphQLSchema createSchema() {
                     builder.dataFetcher("JobReschedule", this::rescheduleJob);
-                    builder.dataFetcher("TaskUpdate", this::updateUserTask);
-                    builder.dataFetcher("TaskPartialUpdate", this::partialUpdateUserTask);
+                    builder.dataFetcher("UserTaskInstanceUpdate", this::updateUserTask);
```
this::updateUserTask -> this::updateUserTaskInstance ?

radtriste commented on data-index/data-index-service/data-index-service-common/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:
```diff
@@ -289,21 +290,30 @@ private CompletableFuture getUserTaskSchema(DataFetchingEnvironment env) {
                 env.getArgument("user"),
                 env.getArgument("groups"),
-                env.getArgument("taskInfo"));
+                env.getArguments());
```
all arguments from env ?

nmirasch replied:
@radtriste  Here We have moved the task related arguments to update to upper level, removing the previous userTaskInfoArgument, letting the operation to pass just the fields that need to be updated. Here I'm passing all the arguments in a map and the operation itself will traduce that to updates if the parameter matches with any 'updatable' field. The schema is in charge to control that only those fields can be passed as arguments ( description, priority,..all the ones that before were contained in UserTaskInfoArgument )

radtriste commented on data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls:
```diff
@@ -356,17 +359,6 @@ input UserTaskInstanceMetaArgument {
     started: DateArgument
     referenceName: StringArgument
 }
-input UserTaskInfoArgument {
```
why removing that structure ?

radtriste reviewed (approved):
lgtm
